File: motor3d 2.0.ConvertedToC#/PosicionRelativa2D.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;


namespace Motor3D.Espacio2D
{
	public enum TipoPosicionRelativa2D
	{
		Secante,
		Coincidente,
		Paralelo
	}

	public class PosicionRelativa2D : ObjetoGeometrico2D
	{

		private TipoPosicionRelativa2D mTipo;

		private Punto2D mInterseccion;
		public TipoPosicionRelativa2D Tipo {
			get { return mTipo; }
		}

		public Punto2D Interseccion {
			get {
				if (mTipo == TipoPosicionRelativa2D.Secante) {
					return mInterseccion;
				} else {
					throw new ExcepcionGeometrica2D("POSICIONRELATIVA2D (INTERSECCION_GET): La posición no es de tipo secante. No se puede obtener la intersección." + Constants.vbNewLine + "Tipo=" + mTipo.ToString());
				}
			}
		}

		public PosicionRelativa2D(Punto2D ValInterseccion)
		{
			mTipo = TipoPosicionRelativa2D.Secante;
			mInterseccion = ValInterseccion;
		}

		public PosicionRelativa2D(TipoPosicionRelativa2D ValTipoPosicion)
		{
			mTipo = ValTipoPosicion;
		}

		public override string ToString()
		{
			if (mTipo == TipoPosicionRelativa2D.Secante) {
				return "{" + mTipo.ToString() + ",Intersección=" + mInterseccion.ToString() + "}";
			} else {
				return "{" + mTipo.ToString() + "}";
			}
		}
	}
}

File: motor3d 2.0/posicionrelativa2d.vb


Namespace Espacio2D
    Public Enum TipoPosicionRelativa2D
        Secante
        Coincidente
        Paralelo
    End Enum

    Public Class PosicionRelativa2D
        Inherits ObjetoGeometrico2D

        Private mTipo As TipoPosicionRelativa2D
        Private mInterseccion As Punto2D

        Public ReadOnly Property Tipo() As TipoPosicionRelativa2D
            Get
                Return mTipo
            End Get
        End Property

        Public ReadOnly Property Interseccion() As Punto2D
            Get
                If mTipo = TipoPosicionRelativa2D.Secante Then
                    Return mInterseccion
                Else
                    Throw New ExcepcionGeometrica2D("POSICIONRELATIVA2D (INTERSECCION_GET): La posición no es de tipo secante. No se puede obtener la intersección." & vbNewLine _
                                                    & "Tipo=" & mTipo.ToString)
                End If
            End Get
        End Property

        Public Sub New(ByVal ValInterseccion As Punto2D)
            mTipo = TipoPosicionRelativa2D.Secante
            mInterseccion = ValInterseccion
        End Sub

        Public Sub New(ByVal ValTipoPosicion As TipoPosicionRelativa2D)
            mTipo = ValTipoPosicion
        End Sub

        Public Overrides Function ToString() As String
            If mTipo = TipoPosicionRelativa2D.Secante Then
                Return "{" & mTipo.ToString & ",Intersección=" & mInterseccion.ToString & "}"
            Else
                Return "{" & mTipo.ToString & "}"
            End If
        End Function
    End Class
End Namespace

